vendor/bat-native-ledger/src/bat_helper.cc

```diff
@@ -2618,6 +2619,25 @@ uint64_t getMediaDuration(const std::map<std::string, std::string>& data,
   return duration;
 }
 
+// static
+bool isValidLeadingUrlSubString(
+    const std::string& url_to_validate,
+    const std::string& to_match) {
+  GURL gurl(url_to_validate);
+  const std::string host_portion =
+    to_match.substr(
+      to_match.find("://") != std::string::npos ? to_match.find("://") + 3 : 0,
+      to_match.find("/"));
+  return gurl.is_valid() &&
+         gurl.spec().find(to_match) != std::string::npos &&
```

Comment:
I thought the whole point was to make sure this isn't part of the query string? `spec` gives you the full url

Reply:
No. We're making sure that the url substring given is actually coming from where we want it to. We have the url that comes in and we want to verify that the substring that contains the hostname comes from that domain and not some other baddomain that could have pathed it in.

Reply:
and if the partial url is a query string for baddomain.com, it won't pass this next line.